supervisor/shared/tick.c

```diff
@@ -102,9 +102,15 @@ void mp_hal_delay_ms(mp_uint_t delay) {
     while (remaining > 0) {
         RUN_BACKGROUND_TASKS;
         // Check to see if we've been CTRL-Ced by autoreload or the user.
-        if(MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)) ||
-           MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_reload_exception)) ||
+        if(MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)))
+           {
+           // clear exception and generate stacktrace
+            MP_STATE_VM(mp_pending_exception) = MP_OBJ_NULL;
+            nlr_raise(mp_obj_new_exception(&mp_type_KeyboardInterrupt));
```

Comment:
Do we need a new exception object or can we just raise the MP_STATE_VM(mp_pending_exception) value?

Reply:
Extraneous #includes have been removed. As far as the exception object, we have to clear the 
MP_STATE_VM(mp_pending_exception) or subsequent sleep() calls return immediately (i.e. the original issue #2865). nlr_raise() cannot be called before clearing the exception, as that will never hit the clear. So, yes, we need a new KeyboardInterrupt exception object.

Reply:
A `MP_STATE_VM(mp_kbd_exception)` object is created once, here:
https://github.com/adafruit/circuitpython/blob/65512cef1c278258f525422fe57271ffac6f51b0/py/runtime.c#L81-L86
So you should just be able to reuse it, I think:
```c
            nlr_raise(&MP_STATE_VM(mp_kbd_exception));
```

Reply:
Yes, testing shows that we can reuse that object. I have changed the code accordingly.